### PR TITLE
refactor: Enhance docs, add tests in `ReverseKGroup`

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java [hunk 1]
```diff
@@ -4,10 +4,7 @@
 import static org.junit.jupiter.api.Assertions.assertNull;
 
 import org.junit.jupiter.api.Test;
-/**
- * Test cases for Reverse K Group LinkedList
- * Author: Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
- */
```
Comment (siriak) [inline src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java:10]: Why did you remove this?

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java (hunk 1 above)
Comment (Hardvan) [reply src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java:10]: I was removing excessive comments from the test code, the above line was deleted by mistake. I added it back

**Review by siriak — APPROVED** [review]
